crates/proof-of-sql/src/sql/parse/query_context_builder.rs

```diff
@@ -79,9 +79,53 @@ impl<'a> QueryContextBuilder<'a> {
         Ok(self)
     }
 
-    pub fn visit_order_by_exprs(mut self, order_by_exprs: Vec<OrderBy>) -> Self {
-        self.context.set_order_by_exprs(order_by_exprs);
-        self
+    pub fn visit_order_by_exprs(
+        mut self,
+        order_by_exprs: Vec<OrderByExpr>,
+    ) -> ConversionResult<Self> {
+        let ident_direction_pairs: Vec<(Ident, bool)> = order_by_exprs
+            .into_iter()
+            .map(|order_by_expr| -> ConversionResult<(Ident, bool)> {
+                let ident_direction_pair = match order_by_expr {
+                    OrderByExpr {
+                        expr: Expr::Identifier(id),
+                        asc: Some(true) | None,
+                        ..
+                    } => Ok((id, true)),
+                    OrderByExpr {
+                        expr: Expr::Identifier(id),
+                        asc: Some(false),
+                        ..
+                    } => Ok((id, false)),
+                    _ => Err(ConversionError::UnsupportedOperation {
+                        message: "Order by columns other than result columns not supported yet"
+                            .to_string(),
+                    }),
+                }?;
+                Ok(ident_direction_pair)
+            })
+            .collect::<ConversionResult<Vec<_>>>()?;
+        // Collect all the result aliases
+        let result_aliases = self
+            .context
+            .get_aliased_result_exprs()?
+            .iter()
+            .map(|aliased_expr| aliased_expr.alias.into())
+            .collect::<IndexSet<Ident>>();
+        let index_direction_pairs: Vec<(usize, bool)> = ident_direction_pairs
+            .into_iter()
+            .map(|ident_direction_pair| -> ConversionResult<(usize, bool)> {
+                let ident = ident_direction_pair.0;
+                let index = result_aliases.get_index_of(&ident).ok_or(
+                    ConversionError::MissingColumnWithoutTable {
+                        identifier: Box::new(ident),
+                    },
+                )?;
+                Ok((index, ident_direction_pair.1))
```

Comment:
NIT: slightly prettier with destructuring the input
```suggestion
            .map(|(ident, direction)| -> ConversionResult<(usize, bool)> {
                let index = result_aliases.get_index_of(&ident).ok_or(
                    ConversionError::MissingColumnWithoutTable {
                        identifier: Box::new(ident),
                    },
                )?;
                Ok((index, direction))
```